### PR TITLE
Fix running Direct3DInterop sample

### LINKED ISSUE #5238: Direct3DInteropSample is broken
**Describe the bug**
Running Direct3DInteropSample will result in the following exception:
```
An unhandled exception of type 'Avalonia.Markup.Xaml.XamlLoadException' occurred in Avalonia.Markup.Xaml.dll
No precompiled XAML found for Direct3DInteropSample.App, make sure to specify x:Class and include your XAML file as AvaloniaResource
```

**To Reproduce**
1. Check out Avalonia master
2. Run Direct3DInteropSample

### PR DESCRIPTION
**Pull request by shartte**
## What does the pull request do?
Fixes Direct3DInterop sample not being able to be run, and also adjusts it to be more in line with the other samples (netcoreapp2, changed XAML file extensions)

## What is the current behavior?
See #5238 

## What is the updated/expected behavior with this PR?
The sample will now start again, but still exhibits #5239, which does not seem to be related to the sample itself.
I also reformatted the csproj to 2-spaces indent, as the other sample project files seem to be.

## Fixed issues
Fixes #5238

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5240) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5240) before we can accept your contribution.<br/><br/>:x: shartte [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5240)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=5240) it.</sub>

File: samples/interop/Direct3DInteropSample/Direct3DInteropSample.csproj [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
-    <ItemGroup>
-        <ProjectReference Include="..\..\..\src\Windows\Avalonia.Direct2D1\Avalonia.Direct2D1.csproj" />
-        <ProjectReference Include="..\..\..\src\Windows\Avalonia.Win32\Avalonia.Win32.csproj" />
-        <ProjectReference Include="..\..\MiniMvvm\MiniMvvm.csproj" />
-    </ItemGroup>
-    <Import Project="..\..\..\build\Rx.props" />
-    <Import Project="..\..\..\build\ReferenceCoreLibraries.props" />
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (FoggyFinder) [inline samples/interop/Direct3DInteropSample/Direct3DInteropSample.csproj:4]: why `2.0`? Other examples use `netcoreapp3.1`

Comment (shartte) [reply samples/interop/Direct3DInteropSample/Direct3DInteropSample.csproj:4]: This is copied from NativeEmbedSample, which uses `<TargetFramework>netcoreapp2.0</TargetFramework>`

Comment (FoggyFinder) [reply samples/interop/Direct3DInteropSample/Direct3DInteropSample.csproj:4]: probably it makes sense to bump it there as well

Comment (shartte) [reply samples/interop/Direct3DInteropSample/Direct3DInteropSample.csproj]: I bumped both to `netcoreapp3.1`.

**Comment by kekekeks** [comment]
>which does not seem to be related to the sample itself.

We are planning to add `DXGIImage` class, so we won't depend on Direct2D for DirectX interop

**Comment by shartte** [comment]
@kekekeks The sample is indeed overall busted because the "draw-under" approach doesn't work anymore apparently. The render-target is cleared by the immediate renderer after the content underneath had been rendered, and then cleared again for the window background.

So is your plan to **remove** the Direct2D backend? If so, I'd not spend any more time on this, but recommend removing this sample as it's completely broken.

**Comment by kekekeks** [comment]
The backend will be kept alive to make sure that we remain backend-independent, but Skia has been the primary one for multiple releases already.

**Comment by shartte** [comment]
If Direct2D will not be removed, I'll investigate why the sample is not actually working (setting all backgrounds to Transparent still clears the render target with a white brush)

**Comment by shartte** [comment]
I will close this for now until a "full" fix is possible without replacing the renderer.

**Pull request closed without merging**